src/main/java/fm/last/moji/impl/FileUploadOutputStream.java

```diff
@@ -90,12 +92,17 @@ public void close() throws IOException {
           delegate.close();
         } finally {
           try {
-            String message = httpConnection.getResponseMessage();
             int code = httpConnection.getResponseCode();
             if (HttpURLConnection.HTTP_OK != code && HttpURLConnection.HTTP_CREATED != code) {
-              throw new IOException(code + " " + message);
-            } else {
-              log.debug("Status: HTTP {} - {}", code, message);
+              InputStream is;
+              if (200 <= httpConnection.getResponseCode() && httpConnection.getResponseCode() <= 299) {
+                is = httpConnection.getInputStream();
+              } else {
+                is = httpConnection.getErrorStream();
+              }
+              String body = IOUtils.toString(is);
```

Comment:
This is missing the encoding so might result in garbage? Is there a way to determine the encoding and pass this through?